Src/IronPython/Runtime/Generator.cs

```diff
@@ -527,7 +527,7 @@ public sealed class PythonGenerator : IEnumerator, IEnumerator<object>, ICodeFor
 
             // This may invoke user code such as __init__, thus MakeException may throw. 
             // Since this is invoked from the generator's body, the generator can catch this exception. 
-            return LightExceptions.Throw(PythonOps.MakeException(Context, throwableBackup[0], throwableBackup[1], throwableBackup[2]));
+            return LightExceptions.Throw(PythonOps.MakeExceptionWorker(Context, throwableBackup[0], throwableBackup[1], throwableBackup[2], false, true));
```

Comment:
By calling the worker directly it looks like `RemoveFrameList` is no longer called (I have no idea what it does).